File: transektcount/src/main/java/com/wmstein/transektcount/widgets/AlertCreateWidget.java
package com.wmstein.transektcount.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.LinearLayout;

import com.wmstein.transektcount.R;

import java.io.Serializable;
import java.util.Objects;

/**************************************************************************
 * This is the widget for creating an alert in the CountOptionsActivity.
 * Created by milo on 02/06/2014.
 * Adopted for TransektCount by wmstein on 18.02.2016
 * Last edited on 2019-02-12
 */
public class AlertCreateWidget extends LinearLayout implements Serializable
{
    EditText alert_name;
    EditText alert_value;
    int alert_id;
    ImageButton deleteButton;

    public AlertCreateWidget(Context context, AttributeSet attrs)
    {
        super(context, attrs);

        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        Objects.requireNonNull(inflater).inflate(R.layout.widget_alert_create, this, true);
        alert_name = findViewById(R.id.alert_name);
        alert_value = findViewById(R.id.alert_value);
        alert_id = 0;
        deleteButton = findViewById(R.id.delete_button);
        deleteButton.setTag(0);
    }

    public String getAlertName()
    {
        return alert_name.getText().toString();
    }

    // this is set to return 0 if it can't parse a value from the box in order
    //   that transektcount doesn't crash
    public int getAlertValue()
    {
        String text = alert_value.getText().toString();
        if (isEmpty(text))
        {
            return 0;
        }
        else
        {
            try
            {
                return Integer.parseInt(text.replaceAll("[\\D]",""));
            } catch (NumberFormatException e)
            {
                return 0;
            }
        }
    }

    public int getAlertId()
    {
        return alert_id;
    }

    public void setAlertName(String name)
    {
        alert_name.setText(name);
    }

    public void setAlertValue(int value)
    {
        alert_value.setText(String.valueOf(value));
    }

    public void setAlertId(int id)
    {
        alert_id = id;
        deleteButton.setTag(id);
    }

    /**
     * Following function is taken from the Apache commons-lang3-3.4 library
     * licensed under Apache License Version 2.0, January 2004
     * <p>
     * Checks if a CharSequence is empty ("") or null.
     * <p>
     * isEmpty(null)      = true
     * isEmpty("")        = true
     * isEmpty(" ")       = false
     * isEmpty("bob")     = false
     * isEmpty("  bob  ") = false
     *
     * @param cs the CharSequence to check, may be null
     * @return {@code true} if the CharSequence is empty or null
     */
    public static boolean isEmpty(final CharSequence cs)
    {
        return cs == null || cs.length() == 0;
    }

}

File: transektcount/src/main/java/com/wmstein/transektcount/widgets/AlertCreateWidget.kt
package com.wmstein.transektcount.widgets

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.EditText
import android.widget.ImageButton
import android.widget.LinearLayout
import com.wmstein.transektcount.R
import java.io.Serializable
import java.util.Objects

/**************************************************************************
 * This is the widget for creating an alert in the CountOptionsActivity.
 * Created by milo on 02/06/2014.
 * Adopted for TransektCount by wmstein on 18.02.2016
 * last edited in Java on 2023-05-09
 * converted to Kotlin on 2023-06-26
 * Last edit on 2023-09-23.
 */
class AlertCreateWidget(context: Context, attrs: AttributeSet?) : LinearLayout(context, attrs),
    Serializable {
    private val alertname: EditText
    private val alertvalue: EditText
    private var alertid: Int
    private val deleteButton: ImageButton

    init {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        Objects.requireNonNull(inflater).inflate(R.layout.widget_alert_create, this, true)
        alertname = findViewById(R.id.alert_name)
        alertvalue = findViewById(R.id.alert_value)
        alertid = 0
        deleteButton = findViewById(R.id.delete_button)
        deleteButton.tag = 0
    }

    var alertName: String?
        get() = alertname.text.toString()
        set(name) {
            alertname.setText(name)
        }

    // this is set to return 0 if it can't parse a value from the box in order
    //   that transektcount doesn't crash
    var alertValue: Int
        get() {
            val text = alertvalue.text.toString()
            return if (isEmpty(text)) {
                0
            } else {
                try {
                    text.replace("\\D".toRegex(), "").toInt()
                } catch (e: NumberFormatException) {
                    0
                }
            }
        }
        set(value) {
            alertvalue.setText(value.toString())
        }
    var alertId: Int
        get() = alertid
        set(id) {
            alertid = id
            deleteButton.tag = id
        }

    companion object {
        /**
         * Following function is taken from the Apache commons-lang3-3.4 library
         * licensed under Apache License Version 2.0, January 2004
         *
         *
         * Checks if a CharSequence is empty ("") or null.
         *
         *
         * isEmpty(null)      = true
         * isEmpty("")        = true
         * isEmpty(" ")       = false
         * isEmpty("bob")     = false
         * isEmpty("  bob  ") = false
         *
         * @param cs the CharSequence to check, may be null
         * @return `true` if the CharSequence is empty or null
         */
        fun isEmpty(cs: CharSequence?): Boolean {
            return cs.isNullOrEmpty()
        }
    }
}